### PR TITLE
Add first-time contributor action

### PR DESCRIPTION
**Pull request by HiDeoo**
#### What kind of changes does this PR include?

- Something else!

#### Description

Following https://github.com/withastro/docs/pull/4966, this pull request adds the same GitHub Actions workflow to the Starlight repository.

Compared to the Astro Docs repository, I made the following changes to the message:

```diff
- Hello! Thank you for opening your **first PR** to Astro’s Docs! 🎉
+ Hello! Thank you for opening your **first PR** to Starlight! 🎉
```

```diff
- If they spot any broken links you will see some error messages on this PR.
+ If they spot any issues you will see some error messages on this PR.
```

```diff
- 2. In a few minutes, you’ll be able to see a preview of your changes on Netlify 🥳
+ 2. In a few minutes, you’ll be able to see a preview of your changes on Vercel 🥳
```

(in preparation of #795)

### CODE REVIEW

**Review by dreyfus92 — APPROVED** [review]
Awesome job @HiDeoo, LGTM ✅

File: .github/workflows/welcome-bot.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        with:
+          repo-token: ${{ secrets.FREDKBOT_GITHUB_TOKEN }}
+          pr-message: |
+            Hello! Thank you for opening your **first PR** to Starlight! 🎉
+            Here’s what will happen next:
+            1. Our GitHub bots will run to check your changes.
+               If they spot any issues you will see some error messages on this PR.
+               Don’t hesitate to ask any questions if you’re not sure what these mean!
+
+            2. In a few minutes, you’ll be able to see a preview of your changes on Vercel 🥳
```
Comment (delucis) [inline .github/workflows/welcome-bot.yml:27]: Let’s actually keep this as Netlify for now. We hit a blocker on #795, so might take a bit of time before we complete the migration.

```suggestion
            2. In a few minutes, you’ll be able to see a preview of your changes on Netlify 🥳
```

File: .github/workflows/welcome-bot.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    name: Welcome First Time Contributors
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/first-interaction@v1
+        with:
+          repo-token: ${{ secrets.FREDKBOT_GITHUB_TOKEN }}
+          pr-message: |
+            Hello! Thank you for opening your **first PR** to Starlight! 🎉
+            Here’s what will happen next:
```
Comment (delucis) [inline .github/workflows/welcome-bot.yml:22]: I think we want newlines either side of this to make a new paragraph:

```suggestion

            Here’s what will happen next:

```

File: .github/workflows/welcome-bot.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  pull_request_target:
+    branches: [main]
+    types: [opened]
+
+permissions:
+  pull-requests: write
+
+jobs:
+  welcome:
+    name: Welcome First Time Contributors
```
Comment (delucis) [inline .github/workflows/welcome-bot.yml:13]: Nit that I got wrong in the docs version 🫣
```suggestion
    name: Welcome First-Time Contributors
```

**Review by delucis — COMMENTED** [review]
Woah that was quick! Thank you @HiDeoo — love it 💖 

Spotted a few tiny details and then should be good to go!

File: .github/workflows/welcome-bot.yml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/first-interaction@v1
+        with:
+          repo-token: ${{ secrets.FREDKBOT_GITHUB_TOKEN }}
+          pr-message: |
+            Hello! Thank you for opening your **first PR** to Starlight! 🎉
```
Comment (delucis) [inline .github/workflows/welcome-bot.yml:21]:
```suggestion
            Hello! Thank you for opening your **first PR** to Starlight! 🌟 
```

**Review by delucis — COMMENTED** [review]
Ooh, actually, let’s also make the emojis Starlight themed 🤩

**Comment by HiDeoo** [comment]
Thanks for the review, not sure where these newlines went between the copy and the PR ^^ Love the ✨ idea.

**Review by delucis — APPROVED** [review]
LGTM! Let’s start welcoming those first-time peeps!

**Pull request merged**